thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/client/thrift/THttpClientDelegate.java

```diff
@@ -309,9 +309,15 @@ private static void handleException(ClientRequestContext ctx, CompletableRpcResp
     }
 
     private static void handlePreDecodeException(ClientRequestContext ctx, CompletableRpcResponse reply,
-                                                 ThriftFunction thriftMethod, Throwable cause) {
-        handleException(ctx, reply, null,
-                        decodeException(cause, thriftMethod.declaredExceptions()));
+                                                 ThriftFunction thriftMethod, Throwable cause,
+                                                 @Nullable HttpRequest request) {
+        final Exception decodedCause =
+                decodeException(cause, thriftMethod.declaredExceptions());
+        if (request != null) {
+            // abort the request to potentially release pooled buffers
+            request.abort(decodedCause);
```

Comment:
This approach seems good and straightforward. However, it will not be able to cover all leaks if a request whose body is a pooled object is rejected by a decorator.

Suggestion) As a separate issue, what do you think of adding a hook to `UserClient` to abort a request if the response is exceptionally completed?

Reply:
> However, it will not be able to cover all leaks where a request whose body is a pooled object is rejected by a decorator.

I think you're probably right - I doubt there is a use case for sending data over even after the response is exceptionally completed. Did you want to create the issue or should I?

Reply:
Created one: https://github.com/line/armeria/issues/5563